src/Mapbender/WmsBundle/Controller/RepositoryController.php

```diff
@@ -221,18 +225,18 @@ public function updateformAction($sourceId)
     public function updateAction($sourceId)
     {
         $request         = $this->get('request');
-        $source          = $this->getDoctrine()->getRepository("MapbenderCoreBundle:Source")->find($sourceId);
+        $source          = $this->loadEntityByPk("MapbenderCoreBundle:Source", $sourceId);
```

Comment:
Maybe better to create and use `getSourceById($sourceId)` method?

Reply:
A simple (no arguments) ```->find()``` on the repository, as used here, actually uses doctrine metadata magic to auto-detect the primary key, which may not always be the id.